assembl/graphql/vote_session.py

```diff
@@ -545,10 +623,117 @@ def mutate(root, args, context, info):
                 vote_spec, 'title', title_entries)
             update_langstring_from_input_entries(
                 vote_spec, 'instructions', instructions_entries)
+            proposal_id = args.get('proposal_id')
+            if proposal_id:
+                proposal_id = int(Node.from_global_id(proposal_id)[1])
+                proposal = models.Idea.get(proposal_id)
+                vote_spec.criterion_idea = proposal
+
             vote_spec.minimum = args['minimum']
             vote_spec.maximum = args['maximum']
             vote_spec.nb_ticks = args['nb_ticks']
             vote_spec.unit = args['unit']
             db.flush()
 
         return UpdateNumberGaugeVoteSpecification(vote_specification=vote_spec)
+
+
+class CreateProposal(graphene.Mutation):
+
+    class Input:
+        vote_session_id = graphene.ID(required=True)
+        title_entries = graphene.List(LangStringEntryInput, required=True)
+        description_entries = graphene.List(LangStringEntryInput, required=True)
+        order = graphene.Float()
+
+    proposal = graphene.Field(lambda: Idea)
+
+    @staticmethod
+    @abort_transaction_on_exception
+    def mutate(root, args, context, info):
+        cls = models.Idea
+        require_cls_permission(CrudPermissions.CREATE, cls, context)
+        vote_session_id = args.get('vote_session_id')
+        vote_session_id = int(Node.from_global_id(vote_session_id)[1])
+        title_entries = args.get('title_entries')
+        description_entries = args.get('description_entries')
+        discussion_id = context.matchdict['discussion_id']
+        discussion = models.Discussion.get(discussion_id)
+
+        with cls.default_db.no_autoflush as db:
+            title_ls = langstring_from_input_entries(title_entries)
+            description_ls = langstring_from_input_entries(description_entries)
+            proposal = cls(
+                discussion_id=discussion_id,
+                title=title_ls,
+                description=description_ls
+            )
+            db.add(proposal)
+            identifier = 'voteSession{}'.format(vote_session_id)
+            root_thematic = get_root_thematic_for_phase(discussion, identifier)
+            if root_thematic is None:
+                root_thematic = create_root_thematic(discussion, identifier)
```

Comment:
So you HAVE to create a proposal in order for the root thematic for vote session to exist? Are proposals now the "questions" that are asked in each line, where tokens/gauges can be associated to?

Reply:
proposal are actually Idea objects (not Question) inside a hidden Thematic (identifier field is only on Thematic)